keanu-project/src/main/java/io/improbable/keanu/network/ModelComposition.java

```diff
@@ -0,0 +1,98 @@
+package io.improbable.keanu.network;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.improbable.keanu.vertices.ProxyVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexId;
+import io.improbable.keanu.vertices.VertexLabel;
+
+public final class ModelComposition {
+
+    private ModelComposition() { }
+
+    /**
+     * Compose one model within a wider model.
+     * <p>
+     *     This function will take a previously constructed BayesNet and do a number of things:
+     *     - Hook up any Proxy Vertices to the parents specified in the inputVertices
+     *     - Increment the depth for all "internal" nodes and the BayesNet itself
+     *     - Pass back any specified output nodes from the Bayesnet (keeping them at the outer depth)
+     *
+     *     Output nodes will be returned unlabelled
+     * </p>
+     * @param bayesianNetwork The Bayesian Network to compose in to the current model
+     * @param inputVertices The mapping from Proxy label to actual input vertex
+     * @param desiredOutputs The set of labels we wish to output from the supplied BayesNet
+     * @return A map of Labels to Output Vertices
+     */
+    public static Map<VertexLabel, Vertex> createModelVertices(BayesianNetwork bayesianNetwork,
```

Comment:
Can we change the name to reflect what it's actually doing, something like `composeModel` or `connectModels` or `connectVertices`